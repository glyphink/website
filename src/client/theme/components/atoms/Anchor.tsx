/**
 * @file Components - Anchor
 * @module website/theme/components/atoms/Anchor
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type FRC,
  type JSX
} from 'react'
import type AnchorProps from './Anchor.props.mts'

/**
 * A hyperlink to web pages, files, email addresses, locations in the same page,
 * or anything else a URL can address.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/a
 * @see {@linkcode AnchorProps}
 * @see {@linkcode FRC}
 *
 * @category
 *  atoms
 *
 * @const {FRC<AnchorProps>} Anchor
 */
const Anchor: FRC<AnchorProps> = forwardRef<HTMLAnchorElement, AnchorProps>(
  /**
   * @param {Props<AnchorProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLAnchorElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<a>` element
   */
  (
    props: Props<AnchorProps>,
    ref: ForwardedRef<HTMLAnchorElement>
  ): JSX.Element => { // eslint-disable-next-line jsx-a11y/anchor-has-content
    return <a {...props} ref={ref} target={props.target ?? '_self'} />
  }
)

Anchor.displayName = 'Anchor'

export default Anchor
