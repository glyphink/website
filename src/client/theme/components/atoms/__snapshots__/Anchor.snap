// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Anchor > should render <a> 1`] = `
<a
  href="https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/a"
  target="_self"
>
  &lt;a&gt;: The Anchor element
</a>
`;
